### PR TITLE
Multilora regression test (compile mode) fix

### PR DESCRIPTION
**Pull request by rsshaik1**
This PR adds dynamo graph breaks after each decoder layer to fix the regression (incorrect output) in multi-lora unit test.

### CODE REVIEW

**Comment by vivekgoe** [comment]
@jczaja @AleksanderStankiewicz please review this change (since this is related to torch.compile mode).  @michalkuligowski please review.

**Comment by michalkuligowski** [comment]
Hi @RonBenMosheHabana Trigger Jenkins Tests  fails on this, please check

**Review by jczaja — COMMENTED** [review]
The change in itself (code) is correct, but the question here is what is the performance impact on our torch compile models : llama-3.1 B8, llama-2-7B  bf16 . So before merging it would be good to test performance throughput and accuracy on our models.

File: vllm/model_executor/models/llama.py [hunk 1]
```diff
@@ -352,7 +352,10 @@ def forward(
                                             kv_caches[i - self.start_layer],
                                             attn_metadata, residual)
             if is_hpu and i % self.config_hidden_layers == 0:
-                htorch.core.mark_step()
+                if htorch.utils.internal.is_lazy():
+                    htorch.core.mark_step()
+                else:
+                    torch._dynamo.graph_break()
```
[the remark this answers is not in the document]
Comment (rsshaik1) [reply vllm/model_executor/models/llama.py:358]: In eager mode the test works as expected, so the graph break might be an empty call.
Could you specify the test to be execute so we can use the same for evaluating performance metrics?

[the remark this answers is not in the document]
Comment (anko-intel) [reply vllm/model_executor/models/llama.py:358]: The test:
              PT_HPU_LAZY_MODE=0  python -u vllm-fork/benchmarks/benchmark_throughput.py \
                        --model /mnt/weka/data/pytorch/llama2/Llama-2-7b-hf \
                        --device hpu \
                        --seed 2024 \
                        --backend vllm \
                        --dataset /software/users/abala/ShareGPT_V3_unfiltered_cleaned_split.json \
                        --num-prompts 1000 \
                        --dtype bfloat16 \
                        --max-model-len 4096 \
                        --max-num-batched-tokens 8192 \
                        --max-num-seqs 128 \
                        --use-padding-aware-scheduling 
But also @bsochack is opposite to use graph_break as rather dubug and internal pytorch API  with other consequences.
BDW, we are now working on adding regional compilation (https://pytorch.org/tutorials/recipes/regional_compilation.html) which also  probably fix your issue ( as I understand is related to very long graph)

[the remark this answers is not in the document]
Comment (vivekgoe) [reply vllm/model_executor/models/llama.py:358]: @anko-intel you are right, we do see performance drop with this change. Since issue we are trying to fix is related to long graph, we would like to understand when we can expect regional compilation to be available, is there a PR for this change which can try out?

[the remark this answers is not in the document]
Comment (anko-intel) [reply vllm/model_executor/models/llama.py:358]: I hope within 2 weeks we can finish it. We  don't have PR yet. For now you can take for test this commit: https://github.com/Kacper-Pietkun/vllm-fork/commit/1dc2e7ee64313fd27c96160f16f3cac7da6285c2 
You can notice that increasing the cache_size also fix performance in solution with graph breaks, but as I previously have written we would like to avoid using  _dynamo,graph_break().
Probably also increasing cache size 3 times will be enough ( in final version it probably will be not needed at all)

[the remark this answers is not in the document]
Comment (bsochack) [reply vllm/model_executor/models/llama.py:358]: just fyi, the target solution should be regional compilation instead of inserting graph_breaks as it can have many unexpected side-effects in torch.compile.
I'm not objecting to merge this change temporally as long as we will replace it with regional compilation and it helps other teams to progress.

[the remark this answers is not in the document]
Comment (anko-intel) [reply vllm/model_executor/models/llama.py:358]: @vivekgoe : regional compilation https://github.com/HabanaAI/vllm-fork/pull/576 is ready to merge

**Comment by michalkuligowski** [comment]
Hi @rsshaik1 regional compilation is merged, lets close this one
